sydent/http/servlets/__init__.py

```diff
@@ -60,13 +67,21 @@ def get_args(request, required_args):
     if len(missing) > 0:
         request.setResponseCode(400)
         msg = "Missing parameters: "+(",".join(missing))
-        return {'errcode': 'M_MISSING_PARAMS', 'error': msg}, None
+        raise MatrixRestError(400, 'M_MISSING_PARAMS', msg)
 
-    return None, args
+    return args
 
 def jsonwrap(f):
     def inner(*args, **kwargs):
-        return json.dumps(f(*args, **kwargs)).encode("UTF-8")
+        try:
+            return json.dumps(f(*args, **kwargs)).encode("UTF-8")
+        except MatrixRestError as e:
+            request = args[1]
```

Comment:
Instead of doing this, can we do `def inner(self, request, *args, **kwargs)` please?